### DISCUSSION TITLE
HOWTO - Use Flowise to Store Chatflows in Postgres OVER SSL?

### DISCUSSION THREAD

**Discussion by dkindlund** [FlowiseAI/Flowise#1617] [Q&A] [answered]
Based on current Flowise documentation, it seems like Flowise can *only* connect to a remote Postgres DB for managing configurations, chatflows, and credentials if and only if Postgres is accepting NON-SSL connections:

```
DATABASE_TYPE=postgres
DATABASE_PORT=5432
DATABASE_HOST=localhost
DATABASE_NAME=flowise
DATABASE_USER=user
DATABASE_PASSWORD=123
```

It feels like we need some sort of additional environment variable like `DATABASE_TLS=true` to instruct Flowise to connect to remote Postgres DBs over TLS/SSL correctly.

Plus, we also need some way to instruct Flowise to either ignore certificate validation OR trust the DB certificate somehow -- likely as additional environment variables.

@HenryHengZJ , any thoughts about this?  To be crystal clear, I'm not talking about using Postgres as a Vector DB store -- I'm just talking about Flowise using Postgres DB to store and manage chatflow settings.

Ref:
https://docs.flowiseai.com/configuration/databases

**Comment by Jaredude** [comment, marked as answer]
You currently can set the env var DATABASE_SSL=true to connect to a Postgres DB that requires SSL (I believe Postgres 15+ defaults to required).

There's also a [PR](https://github.com/FlowiseAI/Flowise/pull/1613) that allows additional options, including self signed certs with the cert as base64 as an env var and the full set of options allowed by the pg npm library (PGSSLMODE).

**Reply by dkindlund** [reply]
Wow, thanks for the pointer @Jaredude -- it was difficult for me to sift through all the different documentations, because of how Flowise can use Postgres in many different ways, like:
* as a configuration store
* as a vector store
* ... etc ...

I'll take a closer look at your PR -- thanks so much!
